public/app/core/components/PageHeader/PageHeader.tsx

```diff
@@ -10,7 +10,7 @@ export interface Props {
 }
 
 const SelectNav = ({ main, customCss }: { main: NavModelItem; customCss: string }) => {
-  const defaultSelectedItem = main.children.find(navItem => {
+  const defaultSelectedItem = main.children?.find(navItem => {
```

Comment:
is this correct? would this not be a null exception if children can be null here?

Reply:
@torkelo Yep, seems to work properly. As children can be null there, because this field is set to be optional in NavModelItem, we don't impact the runtime, just remove the strict-null error (because typescript says that `find` method cannot be run on null/undefined).

Reply:
but if was null here it would mean we would have a null exception? Sure the type is correct?  

If that is the case then the code should be 

const children  = main.children; 
if (!children)  {
  return null; 
}

Reply:
@torkelo we don't, because if `main.children` is null/undefined, within a given PageHeader there aren't any navigation items in both SelectNav and TabsBar, and therefore no tab can be marked as selected. All good, nothing really changes other than the count of strict-null errors.

Reply:
Yeah but in that case we are just showing some empty wrappers don't we? So this may be technically correct, but seems it may be better to make `main.children` not optional cause it does not make sense to render this component in that case. Not sure though how that will ripple in other parts of the code.

Reply:
yes agree @aocenas !

Fixing strict null errors is tricky as it many time highlights problems in the code or the types and many times require a code refactor or a type change

Reply:
@aocenas I don't think we render empty wrappers when `main.children` is undefined/null, because its parent component (Navigation) doesn't render then. We check that there: https://github.com/grafana/grafana/blob/83608baf2f7eca2d47d71eeb9d2f7eb167c45d05/public/app/core/components/PageHeader/PageHeader.tsx#L155
Still we render an empty wrapper when children is defined and is just an empty array - I'm gonna address that too.

Reply:
@aocenas @torkelo included your suggestions. Thanks for your input!